src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java

```diff
@@ -146,11 +147,17 @@ public void process(BlobClient blobClient) {
         }
     }
 
-    private byte[] tryToDownloadBlob(BlobClient blobClient) throws IOException {
+    private byte[] downloadBlob(BlobClient blobClient) throws IOException {
         try (var outputStream = new ByteArrayOutputStream()) {
             blobClient.download(outputStream);
 
             return outputStream.toByteArray();
+        } catch (Exception exc) {
+            String errorMessage = exc instanceof IOException
```

Comment:
We could extend this condition by checking if the cause of this `IOException` is `BlobStorageException` with status code 502. Not sure we should go that deep. Thoughts?

Reply:
Made some changes here, as I used logs from bulk-scan-processor, which uses an older version of the lib. Catching `BlobStorageException` now, after verifying the behaviour in blob-router.

Reply:
Made some changes here, as I used logs from bulk-scan-processor, which uses an older version of the lib. Catching `BlobStorageException` now, after verifying the behaviour in blob-router.